nvm.sh

```diff
@@ -275,6 +275,357 @@ nvm_find_up() {
+            # current_remote_node_version is less and or equal to the current comparator version number so this comparator is satisfied.
+            node_version_compatible=$TRUE
+          else
+            node_version_compatible=$FALSE
+          fi
+
+        elif nvm_string_contains_regexp "$comparator" '^>=[0-9]+\.[0-9]+\.[0-9]+$'; then
+          if nvm_version_greater_than_or_equal_to "$current_remote_node_version" "$stripped_version_from_comparator"; then
+            node_version_compatible=$TRUE
+          else
+            node_version_compatible=$FALSE
+            no_remote_version_will_satisfy_all_comparators=$TRUE
+          fi
+
+        elif nvm_string_contains_regexp "$comparator" '^<[0-9]+\.[0-9]+\.[0-9]+$'; then
+          if nvm_version_greater "$stripped_version_from_comparator" "$current_remote_node_version"; then
+            node_version_compatible=$TRUE
+          else
+            node_version_compatible=$FALSE
+          fi
+
+        elif nvm_string_contains_regexp "$comparator" '^>[0-9]+\.[0-9]+\.[0-9]+$'; then
+          if nvm_version_greater "$current_remote_node_version" "$stripped_version_from_comparator"; then
+            node_version_compatible=$TRUE
+          else
+            node_version_compatible=$FALSE
+            no_remote_version_will_satisfy_all_comparators=$TRUE
+          fi
+
+        else
+          node_version_compatible=$FALSE
+        fi
+        # stop checking if all comparators are compatible with current_remote_node_version upon finding one that is incompatible
+        [ $node_version_compatible -eq $FALSE ] && comparator_set_copy=''
+      done # while [ -n "$comparator_set_copy" ]; do
+      # stop iterating through remote_node_versions_copy for this comparator_set upon finding the first remote node version that is compatible with all comparators.
+      [ $node_version_compatible -eq $TRUE ] && highest_compatible_node_versions="$highest_compatible_node_versions $current_remote_node_version"
+    done # while [ -n "$remote_node_versions_copy" ] && [ $node_version_compatible -eq $FALSE ] && [ $no_remote_version_will_satisfy_all_comparators -eq $FALSE ]; do
+
+  done
+
+  highest_compatible_node_version='0.0.0'
+
+  highest_compatible_node_versions=$(command printf "%s" "$highest_compatible_node_versions" | command tr ' ' '\n')
+  while [ -n "$highest_compatible_node_versions" ]; do
+    compatible_node_version=$(command printf "%s" "$highest_compatible_node_versions" | head -n1 | command sed -E 's/^ +//;s/ +$//')
+    highest_compatible_node_versions=$(command printf "%s" "$highest_compatible_node_versions" | tail -n +2)
+    [ -n "$compatible_node_version" ] || continue
+
+    if nvm_version_greater "$compatible_node_version" "$highest_compatible_node_version"; then
+      highest_compatible_node_version="$compatible_node_version"
+    fi
+  done
+  [ "$highest_compatible_node_version" != '0.0.0' ] && nvm_echo "$highest_compatible_node_version"
+) }
+
+nvm_find_package_json() { (
```

Comment:
why are the parens here wrapping the function body? (same with all the functions)

Reply:
I'd prefer to keep the current format of the project for now; if we want to explore using that pattern, it'd need to be a separate PR.